tensorflow_serving/batching/batching_util.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_SERVING_BATCHING_BATCHING_UTIL_H_
+#define TENSORFLOW_SERVING_BATCHING_BATCHING_UTIL_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/lib/core/status.h"
+
+
+
+namespace tensorflow {
+namespace serving {
+
+// For batch of inputs calculates maximum dim sizes across all tensors
+// with the same name.
+// These dim sizes are used later to calculate padding amount for each tensor.
+// For example, for batch containing three tasks with the following inputs
+// (instead of tensors there are their shapes):
+//
+// task1: {'features': [100, 500, 300], 'true_labels': [100]}
```

Comment:
I found "features" and "true_labels" confusing and distracting. Make it fully abstract and just use "tensor_a" and "tensor_b"?